enhancements/sig-architecture/64-placementStrategy/README.md

```diff
@@ -0,0 +1,360 @@
+
+### Non-Goals
+
+1. The new Placement APIs field placementStrategy has no control over applying the workloads.
+1. The new Placement APIs field placementStrategy has no control over scheduling the workloads.
+
+## Proposal
+
+### User Stories
+
+#### Story 1: As end user I want to define a subset of the selected clusters as a affinity clusters.
+#### Story 2: As end user I want to divide the selected clusters to groups then a workloads applier APIs can use those cluster groups to roll out the workloads. 
+
+### Placement placementStrategy API
+
+PlacementStrategy's ClustersGroupPercentage field define a percentage of the total selected clusters that will divide the clusters to groups(batches). As example; for a total 100 clusters selected, clustersGroupPercentage equal to 20% will divide the selected clusters into 5 groups each group consist of 20 clusters. ClustersGroupPercentage default value is None. NumberOfClustersPerGroup is the max number of managedClusters can be exist in a single cluster group. ClustersGroupPercentage and NumberOfClustersPerGroup cannot be used at the same time. If ClustersGroupPercentage is defined it will override the NumberOfClustersPerGroup. AffinityClusterGroup is used to identify a subset of the managedClusters. The AffinityClusterGroup has its own label selector and affinityGroupPercentage.
+
+```go
+type PlacementSpec struct {
+	// ClusterSets represent the ManagedClusterSets from which the ManagedClusters are selected.
+	// +optional
+	ClusterSets []string `json:"clusterSets,omitempty"`
+
+	// Predicates represent a slice of predicates to select ManagedClusters. The predicates are ORed.
+	// +optional
+	Predicates []ClusterPredicate `json:"predicates,omitempty"`
+            ...
+ 
+	// PlacementStrategy define affinity clusters as subset of the selected clusters and divide the selected clusters to groups.
+	// +optional
+	PlacementStrategy PlacementStrategy `json:"placementStrategy,omitempty"`
+}
+
+type AffinityClusterGroup struct {
+	// LabelSelector represents a selector of ManagedClusters by label
+	// +optional
+	ClusterSelector metav1.LabelSelector `json:"clusterSelector,omitempty"`
+
+	// ClusterGroupPercentage is a percentage of the total selected clusters that will divide the clusters to groups.
+	// ex; for a total 100 clusters selected, ClusterGroupPercentage equal to 20% will divide the selected clusters to
+	//     5 groups each group should have 20 clusters.
+	// +kubebuilder:validation:Enum="5%";"10%";"15%";"20%";"25%";"30%";"50%";"100%"
+	// +kubebuilder:default:=100%
+	// +optional
+	AffinityGroupPercentage string `json:"affinityGroupPercentage,omitempty"`
+}
+
+type PlacementStrategy struct {
+	// Affinity clusters label selector. Affinity clusters will be placed in cluster group index 0 and it will have MandatoryClusterGroup type.
+	// +optional
+	AffinityClusterGroup AffinityClusterGroup `json:"affinityClusterGroup,omitempty"`
```

Comment:
Instead of having an AffinityClusterGroup here. I think we should define two decision generator, one is static with a predefined label, and another is by percentage e.g.

```yaml
decisionGenerators:
- type: static
  static:
    labels:
      # this will add labels in related decision
      purpose: canary
    clusterSelector:
      purpose: canary
- type: byGroup
  byGroup:
    numberOfClustersPerGroup: 150
```

Reply:
So as the suggested list , is it possible to have two items with static type ? how about the byGroup type ? can we also have 2 items with byGroup type ?

Reply:
I think we can have multiple static groups but we can only have one byGroup type and the byGroup type needs to be the last item in the list.